etc/listr2.api.md

```diff
@@ -245,17 +245,17 @@ export class ListrTaskObject<Ctx, Renderer extends ListrRendererFactory> extends
         count: number;
         withError?: any;
     };
-    run(context: Ctx, wrapper: ListrTaskWrapper<Ctx, Renderer>): Promise<void>;
-    skip: boolean | string | ((ctx: Ctx) => boolean | string | Promise<boolean> | Promise<string>);
+    run(context: Ctx | undefined, wrapper: ListrTaskWrapper<Ctx, Renderer>): Promise<void>;
+    skip: boolean | string | ((ctx: Ctx | undefined) => boolean | string | Promise<boolean> | Promise<string>);
     // (undocumented)
     set state$(state: ListrTaskState);
-    state: string;
-    subtasks: ListrTaskObject<Ctx, any>[];
-    task: (ctx: Ctx, task: ListrTaskWrapper<Ctx, Renderer>) => void | ListrTaskResult<Ctx>;
+    state?: string;
+    subtasks?: ListrTaskObject<Ctx, any>[];
+    task: (ctx: Ctx | undefined, task: ListrTaskWrapper<Ctx, Renderer>) => void | ListrTaskResult<Ctx>;

```

Comment:
These could have been `undefined` all along, due to `strictNullChecks` not being enabled yet

Reply:
Context is always defined as an empty object initially, so it should not be a problem.

Reply:
Thanks for this info. When I get a chance to take another pass, I'll make it required everywhere